### PR TITLE
use more OTP versions and switch to GH actions

### PR DESCRIPTION
**Pull request by RoadRunnr**

### CODE REVIEW

**Comment by RoadRunnr** [comment]
draft review to collect some feedback.

The conversion to GH actions loses the slack notifications after CI builds. There are GH actions available to achieve something similar to Travis Slack integration, but they are much more complex (too complex IMHO).

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    types:
+      - created
+
+jobs:
+  test:
+    name: CI
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        otp: [21.3.8.18, 22.0.7, 22.1.8, 22.2.8, 22.3.4.14, 23.0.4, 23.1.5.0, 23.2.1.0]
```
Comment (ebengt) [inline .github/workflows/main.yml:17]: Are all the minor versions needed?

**Review by ebengt — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (RoadRunnr) [reply .github/workflows/main.yml:17]: yes and no, I have locally some problems with a specific version. The lists makes sure that all recent minor version work.
But we might indeed reduce this a bit.

**Review by vkatsuba — COMMENTED** [review]
Looks like if Travis will removed we also don't need `rebar.config.script` file.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+           apk update
+           apk --no-cache upgrade
+           apk --no-cache add gcc git libc-dev libc-utils libgcc linux-headers make bash musl-dev musl-utils ncurses-dev pcre2 pkgconf scanelf wget zlib
+    - name: Build
+      run: rebar3 compile
+    - name: Run tests
+      run: |
+           rebar3 do xref
+           rebar3 do ct
+           rebar3 as test do coveralls send || /bin/true
```
Comment (RoadRunnr) [inline .github/workflows/main.yml:33]: coverall on GH actions requires rebar.config.script changes. Also, running a matrix test requires the coverall parallel option and the end webhook

Comment (vkatsuba) [reply .github/workflows/main.yml:33]: Done

**Pull request merged**